SetUserSessionLockActivity/handler.ts

```diff
@@ -0,0 +1,176 @@
+/**
+ * Interacts with Session API to lock/unlock user
+ */
+
+import { Context } from "@azure/functions";
+import { toError } from "fp-ts/lib/Either";
+import {
+  fromEither,
+  fromLeft,
+  taskEither,
+  TaskEither,
+  tryCatch
+} from "fp-ts/lib/TaskEither";
+import * as t from "io-ts";
+import { readableReport } from "italia-ts-commons/lib/reporters";
+import { FiscalCode } from "italia-ts-commons/lib/strings";
+import { SuccessResponse } from "../generated/session-api/SuccessResponse";
+import { Client } from "../utils/sessionApiClient";
+
+function assertNever(_: never): void {
+  throw new Error("should not have executed this");
+}
+
+// Activity input
+export const ActivityInput = t.interface({
+  action: t.union([t.literal("LOCK"), t.literal("UNLOCK")]),
+  fiscalCode: FiscalCode
+});
+export type ActivityInput = t.TypeOf<typeof ActivityInput>;
+
+// Activity result
+export const ActivityResultSuccess = t.interface({
+  kind: t.literal("SUCCESS")
+});
+export type ActivityResultSuccess = t.TypeOf<typeof ActivityResultSuccess>;
+
+// Activity failed because of invalid input
+export const InvalidInputFailure = t.interface({
+  kind: t.literal("INVALID_INPUT_FAILURE"),
+  reason: t.string
+});
+export type InvalidInputFailure = t.TypeOf<typeof InvalidInputFailure>;
+
+// Activity failed because of an error on an api call
+export const ApiCallFailure = t.interface({
+  kind: t.literal("API_CALL_FAILURE"),
+  reason: t.string
+});
+export type ApiCallFailure = t.TypeOf<typeof ApiCallFailure>;
+
+// Activity failed because the api has been called badly
+export const BadApiRequestFailure = t.interface({
+  kind: t.literal("BAD_API_REQUEST_FAILURE"),
+  reason: t.string
+});
+export type BadApiRequestFailure = t.TypeOf<typeof BadApiRequestFailure>;
+
+// maps domain errors that are considered transient and thus may allow a retry
+export const TransientFailure = ApiCallFailure;
+export type TransientFailure = t.TypeOf<typeof TransientFailure>;
+
+export const ActivityResultFailure = t.taggedUnion("kind", [
+  ApiCallFailure,
+  BadApiRequestFailure,
+  InvalidInputFailure
+]);
+export type ActivityResultFailure = t.TypeOf<typeof ActivityResultFailure>;
+
+export const ActivityResult = t.taggedUnion("kind", [
+  ActivityResultSuccess,
+  ActivityResultFailure
+]);
+
+export type ActivityResult = t.TypeOf<typeof ActivityResult>;
+
+const logPrefix = `SetUserSessionLockActivity`;
+
+/**
+ * Wraps the logic to call Session API and lift errors to the correct domain values
+ * @param sessionApiClient
+ * @param action
+ * @param fiscalCode
+ */
+const callSessionApi = (
+  sessionApiClient: Client<"ApiKey">,
+  action: ActivityInput["action"],
+  fiscalCode: FiscalCode
+): TaskEither<ApiCallFailure | BadApiRequestFailure, SuccessResponse> =>
```

Comment:
Can we add a super type for `ApiCallFailure` and `BadApiRequestFailure`

Reply:
We can union such in a type that models "failures that can happen when calling api".
I think it would only make this function more obscure. 
I'd rather use `ActivityResultFailure` to indicate "hey, don't care about the kind of failure that can happen here".